derive/tests/compiletest.rs

```diff
@@ -0,0 +1,14 @@
+mod compile;
+
+/// The `compile_fail` tests in `tests/compile_fail/` are inherently fragile,
+/// as small changes in output from rustc or from the trybuild crate may cause
+/// them to fail. To regenerate the `*.stderr` files, run:
+/// `TRYBUILD=overwrite cargo test -p zk_evm_derive -- test_compile_fail`.
+/// Then, check the git diff to ensure that the new `*.stderr` files are what
+/// you would expect (most importantly, make sure they actually contain errors).
```

Comment:
new issue: if we care about this, we should pin a compiler version in `rust-toolchain.toml`

Reply:
It is pretty minimalistic error output. I have tried compilers 1.65, 1.71 and latest nightly, with the `trybuild`, 1.0.54 and latest (so covering last 2 years), and output hasn't been changed and the tests pass. 

Maybe it is not bad idea to set `rust-toolchain.toml` for the project but for other reasons (few times happened to me that the build breaks because I didn't have latest nightly, or that build breaks because compiler is above some nighlty version). If you thought to set `rust-tooclhain.toml` just for `derive` folder, that is also fine (and works).

Reply:
We could also ignore the compile-fail tests, or run them with `TRYBUILD=overwrite` in the CI. This would basically trade a potentially flaky CI for tests that might pass in the CI but fail locally. Contrary to what might be implied by my comment above, `TRYBUILD=overwrite` still fails if the test doesn't fail to compile, so we wouldn't have tests secretly failing in the CI.

But it sounds like it's a small chance of failure either way (thanks for checking @atanmarko).

Relatedly, I realized there is no `cargo test --workspace` in the CI, so I need to add the derive tests in still.